### PR TITLE
Added options to command find_package

### PR DESCRIPTION
**Pull request by bmarques1995**
Now is possible to use the command find_package with cmake to find nghttp3

### CODE REVIEW

**Comment by tatsuhiro-t** [comment]
@vszakats What do you think about this change?

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,33 @@ set(nghttp3_SOURCES
...
+  
+  # Install cmake config files
+  install(
+      FILES "${NGHTTP3_PROJECT_CONFIG}" "${NGHTTP3_VERSION_CONFIG}"
+      DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
+
+   install(
+    EXPORT "${NGHTTP3_TARGETS_EXPORT_NAME}"
+    NAMESPACE "${NGHTTP3_NAMESPACE}"
+    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
```
Comment (vszakats) [inline lib/CMakeLists.txt:91]:
```suggestion
  write_basic_package_version_file(
    "${NGHTTP3_VERSION_CONFIG}" VERSION ${NGHTTP3_VERSION} COMPATIBILITY SameMajorVersion
  )
  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/config.cmake.in" "${NGHTTP3_PROJECT_CONFIG}" @ONLY)
  
  # Install cmake config files
  install(
    FILES "${NGHTTP3_PROJECT_CONFIG}" "${NGHTTP3_VERSION_CONFIG}"
    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")

  install(
    EXPORT "${NGHTTP3_TARGETS_EXPORT_NAME}"
    NAMESPACE "${NGHTTP3_NAMESPACE}"
    DESTINATION "${NGHTTP3_CONFIG_INSTALL_DIR}")
```
nits: whitespace

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -89,8 +117,8 @@ if(ENABLE_STATIC_LIB)
   target_compile_definitions(nghttp3_static PUBLIC "-DNGHTTP3_STATICLIB")
 
   install(TARGETS nghttp3_static
+    EXPORT ${NGHTTP3_TARGETS_EXPORT_NAME} 
     DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 endif()
 
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libnghttp3.pc"
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+
```
Comment (vszakats) [inline lib/CMakeLists.txt:124]:
```suggestion
```
nit: delete extra newline at the end of file.

File: lib/cmake_modules/config.cmake.in [hunk 3]
```diff
@@ -0,0 +1,3 @@
+include(CMakeFindDependencyMacro)
+
+include("${CMAKE_CURRENT_LIST_DIR}/@NGHTTP3_TARGETS_EXPORT_NAME@.cmake")
```
Comment (vszakats) [inline lib/cmake_modules/config.cmake.in:3]: A newline is missing at he end of this file.

What do you think of moving this file to the existing `cmake` directory instead?
Along with the generated `.cmake` modules above.

**Comment by tatsuhiro-t** [comment]
@vszakats Thank you for the quick review!

File: lib/cmake_modules/config.cmake.in (hunk 3 above)
Comment (bmarques1995) [reply lib/cmake_modules/config.cmake.in:3]: It's okay for me

**Comment by bmarques1995** [comment]
> @vszakats Thank you for the quick review!

I've modified the files, I've moved the pkgconfig to the end of the CMakeLists.txt, where it was located originally, about the modifications suggested by @vszakats, i perceived that the snippets indicated by him was at same identation level of the entry script, meaning without identation. I've moved the config.cmake.in to the same CMakeLists.txt dir.

**Comment by vszakats** [comment]
Thanks @bmarques1995. Looks good to me, also agreeing on the `lib` directory for the new file.

**Comment by tatsuhiro-t** [comment]
lib/config.cmake.in needs to be added to 
https://github.com/ngtcp2/nghttp3/blob/6bcfffb861501c2d1be02fc021e343ccb6456c74/lib/Makefile.am#L26 so that generated archive file contains it.

**Comment by bmarques1995** [comment]
Could u test again? If you approve the modifications i will squash the commits in one

**Comment by tatsuhiro-t** [comment]
Thank you for PR.  Merged now.

**Pull request merged**